services-core/src/gas_price/priority.rs

```diff
@@ -0,0 +1,76 @@
+use super::GasPriceEstimating;
+use anyhow::{anyhow, Result};
+use std::{future::Future, time::Duration};
+
+// Uses the first successful estimator.
+pub struct PriorityGasPrice {
+    estimators: Vec<Box<dyn GasPriceEstimating>>,
+}
+
+impl PriorityGasPrice {
+    pub fn new(estimators: Vec<Box<dyn GasPriceEstimating>>) -> Self {
+        Self { estimators }
+    }
+
+    async fn prioritize<'a, T, F>(&'a self, operation: T) -> Result<f64>
```

Comment:
Totally optional, but seeing how much you like to not repeat code, this could probably be made even more generic by returning the same type as operation. In that case we might be able to use for our other "Priority" components (e.g. price source as well).